File: lua/Janken.lua
local ansList = {"グー", "チョキ", "パー"}

print("最初はグー、じゃんけん・・・")
io.write("あなた：")
You = io.read()

math.randomseed(os.time())
local num = math.random(#ansList)
Lua = ansList[num]
print(("あいて：%s"):format(Lua))
if You == Lua then
  print("あいこです")
elseif You == "グー" and Lua == "チョキ" then
  print("あなたの勝ちです")
elseif You == "グー" and Lua == "パー" then
  print("あなたの負けです")
elseif You == "チョキ" and Lua == "グー" then
  print("あなたの負けです")
elseif You == "チョキ" and Lua == "パー" then
  print("あなたの勝ちです")
elseif You == "パー" and Lua == "グー" then
  print("あなたの勝ちです")
elseif You == "パー" and Lua == "チョキ" then
  print("あなたの負けです")
else
  print("正しいポーズを選んでください")
end

File: Go/Janken.go
package main

import (
  "bufio"
	"fmt"
  "os"
  "math/rand"
  "time"
)

func main() {
  var ansList[3] string = [3]string {"グー", "チョキ", "パー"}

	fmt.Println("最初はグー、じゃんけん・・・")
  fmt.Print("あなた：")

  scanner := bufio.NewScanner(os.Stdin)
  scanner.Scan()
  You := scanner.Text()

  rand.Seed(time.Now().UnixNano())
  num := rand.Intn(len(ansList))
  Go := ansList[num]

  fmt.Println("あいて：" + Go)
  if You == Go {
    fmt.Println("あいこです")
  } else if You == "グー" && Go == "チョキ" {
    fmt.Println("あなたの勝ちです")
  } else if You == "グー" && Go == "パー" {
    fmt.Println("あなたの負けです")
  } else if You == "チョキ" && Go == "グー" {
    fmt.Println("あなたの負けです")
  } else if You == "チョキ" && Go == "パー" {
    fmt.Println("あなたの勝ちです")
  } else if You == "パー" && Go == "グー" {
    fmt.Println("あなたの勝ちです")
  } else if You == "パー" && Go == "チョキ" {
    fmt.Println("あなたの負けです")
  } else {
    fmt.Println("正しいポーズを選んでください")
  }

}
